pkg/templates/sample/report/ReportTesterProcessor.gen.go

```diff
@@ -0,0 +1,20 @@
+package istio_mixer_adapter_sample_report
+
+import "istio.io/mixer/pkg/adapter/config"
+
+const TemplateName = "istio.mixer.adapter.sample.report.Sample"
+
+// Instance represent the runtime structure that will be passed to the ReportSample method in the handlers.
+type Instance struct {
+	Name       string
+	Value      interface{}
+	Dimensions map[string]interface{}
+}
+
+// SampleProcessor represent the Go interface that handlers must implement if it wants to process the template
+// named `istio.mixer.adapter.sample.report.Sample`
+type SampleProcessor interface {
+	config.Handler
+	ConfigureSample(map[string]*Type /*Constructor:instance_name to Type mapping. Note type name will not be passed at all*/) error
+	ReportSample([]*Instance /*The type is inferred from the Instance.name and the mapping of instance to types passed during the config time*/) error
```

Comment:
So, two options:

We pass a getter version of Instance and Type. Which we get ugly since both Type and Instance will be defined in protos, which contains setter too, right ?
If two handlers reference a same constructor.instance_name and both handlers are resolved for a given request path, we make a deep copy of Type and Instance before handing to each handler. Too expensive in terms of performance.
So today, we don't have a problem, because we create a new aspect object for every resolved aspect within a request. If we make a deep copy of a completely evaluated constructor (Instance object), it would be identical to what we are doing now. May be the perf impact is not that bad.. Making a copy of Instance should definitely be better than running expression evaluator for each constructor(current flow).

Given, that **I vote for passing pointers and making a deep copy** if more than one handlers, that are selected at runtime, happens to reference the same Constructor.instance_name